.github/workflows/ci.yml

```diff
@@ -42,3 +42,6 @@ jobs:
         with:
           name: proptest-regressions
```

Comment:
I suggest also run:

```yaml
 - name: Test release build with debug assertions
   run:  cargo test --release
   env:
     CARGO_PROFILE_RELEASE_DEBUG_ASSERTIONS: 1
     CARGO_PROFILE_RELEASE_OVERFLOW_CHECKS: 1
```

and

```yaml
 - name: Test release build with debug assertions
   run:  cargo test --release
```

Reply:
Sometimes release build could optimize `unsafe` code and find out some assumptions you made that is wrong.

Reply:
I also recommend using [`cargo-careful`](https://crates.io/crates/cargo-careful) for testing, you can use [taiki-e/install-action](github.com/taiki-e/install-action) to install it.